addon/util.js

```diff
@@ -26,7 +26,7 @@ export const OWNER_KEY = (function() {
 })();
 
 export function isObject(value) {
```

Comment:
I'm not sure I like the name `isObject` for this util anymore, maybe `isExtantNonArrayObject` ?

Reply:
I had to see what "Extant" means :)

If you elaborate what is your concern of this being named `isObject`, we may be able to find better name.

Reply:
Extant basically just means "exists" or in this case "not null". My main thing is that `isObject` usually does not also check `!isArray` so I'd like to make that clear. And I figured if we are going to go so far as `isNonArrayObject` I might as well slip in a `not array and not null` point to it.

Reply:
I'd also be happy with `isNonArrayObject` though.

Reply:
`Extant` is not quite right.  It has a strong connotation of "still around" which is not really what we're testing here.

`isObject` seems ok if we change the implementation to something like `value && value.constructor === Object`

Basically we're looking for POJOs.

Reply:
I like changing it to `value.constructor === Object`. Any objections? @runspired @hjdivad

Reply:
Checking constructor prevents using Object.create(null) objects, will that be an issue for us?

Do we care about anything other than `typeof obj === ‘object’ && obj !== null`?

Reply:
We care that it's not an array. David and I discussed this OOB this evening and think we should just rename the existing implantation (not null not array type object) to something like "isEmbeddedObject" or some such

Reply:
Basically rename to the reason for the checks instead of what the check is

Reply:
OK, renamed the function to `isEmbeddedObject`, because:
- embedded is already the term we use to refer to the sub-models, which can be part of a model/record
- we use the function in two different places (not in the `merge` function) and although they are similar uses (both determining whether the logic should recurse), they are different to make it difficult to find a name, which fits both cases.